### PR TITLE
chore(example): Add examples

### PR DESCRIPTION
**Pull request by algorithmcardboard**
## Description of this change

> Add example code for implementing scripts that can be executed by Ketch's agent.

## Why is this change being made?
- [ ] Chore (non-functional changes)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested? How can the reviewer verify your testing?

Tested by running the agent flow with dynamic import of the example in my local.

## Checklist

- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have evaluated the security impact of this change, and [OWASP Secure Coding Practices](https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/migrated_content#) have been observed.
- [X] I have informed stakeholders of my changes.

### CODE REVIEW

File: examples/mongodsr/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  } catch (error) {
+    console.error(error)
+  } finally {
+    console.log('in finally')
+    await client.close()
+  }
+  return new Promise<DSRResponse>(resolve => {
+    return {
+      apiVersion: req.apiVersion,
+      kind: 'DeleteStatusEvent',
```
Comment (sethyates) [inline examples/mongodsr/src/index.ts:25]: This is a response not event

File: examples/mongodsr/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      metadata: req.metadata,
+      response: {
+        status: 'completed',
```
Comment (sethyates) [inline examples/mongodsr/src/index.ts:28]: There are enums for this

File: examples/mongodsr/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        reason: 'executed',
```
Comment (sethyates) [inline examples/mongodsr/src/index.ts:29]: There are enums for this

File: examples/mongodsr/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        expectedCompletionTimestamp: Date.now(),
```
Comment (sethyates) [inline examples/mongodsr/src/index.ts:30]: Please check that Date.now() returns the correct resolution per spec

File: examples/mongodsr/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        identities: req.request.identities,
```
Comment (sethyates) [inline examples/mongodsr/src/index.ts:31]: These three haven't changed so shouldn't be returned

File: src/index.ts [hunk 6]
```diff
@@ -15,51 +15,42 @@
...
  */
-export type HandleRequest = (request: DsrRequest, connectionConfig: ConnectionConfig) => Promise<DsrResponse>
+export type HandleRequest = (request: DSRRequest, connectionConfig: ConnectionConfig) => Promise<DSRResponse>
 
 export type DSRVersion = 'dsr/v1'
 
 /**
  * The status of the Data Subject Request.
  */
-export enum RequestStatus {
```
Comment (sethyates) [inline src/index.ts:25]: I think you need to change something in tsconfig to get enums to work rather than drop them

File: src/index.ts [hunk 7]
```diff
@@ -118,9 +109,9 @@ export interface Metadata {
  */
 export interface AccessRequest {
   apiVersion: DSRVersion
-  kind: Kind.AccessRequest
+  kind: Kind
   metadata: Metadata
-  requestBody: {
+  request: {
```
Comment (sethyates) [inline src/index.ts:114]: Several of the request and response fields should be changed to optional (`documents?` For example).

**Review by sethyates — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 7 above)
Comment (algorithmcardboard) [reply src/index.ts:121]: Added `?` for optional fields as per the ketch-forwarder documentation.

File: examples/mongodsr/src/index.ts (hunk 1 above)
Comment (algorithmcardboard) [reply examples/mongodsr/src/index.ts]: Updated.

File: examples/mongodsr/src/index.ts (hunk 2 above)
Comment (algorithmcardboard) [reply examples/mongodsr/src/index.ts]: Updated to use enums.  That was brilliant.

File: examples/mongodsr/src/index.ts (hunk 4 above)
Comment (algorithmcardboard) [reply examples/mongodsr/src/index.ts]: It didn't.  Had to do Math.floor.  Good catch.

File: examples/mongodsr/src/index.ts (hunk 5 above)
Comment (algorithmcardboard) [reply examples/mongodsr/src/index.ts]: removed them

File: src/index.ts (hunk 6 above)
Comment (algorithmcardboard) [reply src/index.ts]:
```
"preserveConstEnums": true,
```
did the trick.

File: examples/mongodsr/src/index.ts [hunk 8]
```diff
@@ -1,48 +1,41 @@
...
     console.log('deleted')
   } catch (error) {
     console.error(error)
   } finally {
     console.log('in finally')
     await client.close()
   }
-  return new Promise<DsrResponse>(resolve => {
-    resolve({
+  return new Promise<DSRResponse>(resolve => {
```
Comment (sethyates) [inline examples/mongodsr/src/index.ts:29]: This is convoluted. Just return the { without the Promise<...resolve...

**Review by sethyates — CHANGES_REQUESTED** [review]

File: examples/mongodsr/src/index.ts (hunk 3 above)
Comment (algorithmcardboard) [reply examples/mongodsr/src/index.ts]: Fixed to use enums.

File: examples/mongodsr/src/index.ts (hunk 8 above)
Comment (algorithmcardboard) [reply examples/mongodsr/src/index.ts]: Fixed.

**Review by sethyates — APPROVED** [review]

**Pull request merged**